R/XcmsExperiment-functions.R

```diff
@@ -1169,6 +1209,70 @@ XcmsExperiment <- function() {
     as(MsExperiment(), "XcmsExperiment")
 }
 
+#' Convert a XCMSnExp to a XcmsExperiment.
+#'
+#' @noRd
+.xcms_n_exp_to_xcms_experiment <- function(from) {
+    requireNamespace("MsExperiment", quietly = TRUE)
+    ## Check requirements:
+    ## - an empty processing queue
+    if (length(from@spectraProcessingQueue))
+        stop("Processing queue is not empty. Can only convert objects with ",
+             "an empty spectra processing queue.")
+    res <- readMsExperiment(spectraFiles = fileNames(from),
+                            sampleData = MSnbase::pData(from))
+    res <- as(res, "XcmsExperiment")
+    res@processHistory <- from@.processHistory
+    res <- filterSpectra(
+        res, filterRt, rt = range(rtime(from, adjusted = FALSE)))
+    if (hasAdjustedRtime(from)) {
+        rts <- rtime(from)
+        if (length(rts) != length(res@spectra))
+            stop("Number of spectra don't match. Was the XCMSnExp subset?")
+        res@spectra$rtime_adjusted <- unname(rts)
+    }
+    if (hasChromPeaks(from)) {
+        res@chromPeaks <- chromPeaks(from)
+        res@chromPeakData <- as.data.frame(chromPeakData(from))
+    }
+    if (hasFeatures(from))
+        res@featureDefinitions <- as.data.frame(featureDefinitions(from))
+    res
+}
+
+#' Combine `XcmsExperiment` objects. Only combining of chrom peaks is supported.
+#' Any alignment or correspondence results are removed.
+#'
+#' @param x `list` of `XcmsExperiment` objects.
+#'
+#' @noRd
+.xmse_combine <- function(x) {
```

Comment:
Again internal, but would be `xcmse` or something else be more intiutive ?

Reply:
again, yes. I was using `.mse_` as an abbreviation for `MsExperiment` in internal function names - to quickly see from the function name that this is supposed to work on `MsExperiment` objects - then for `XcmsExperiment` I used `xmse` - and true, I sometimes even misspell it now. Would just need changes in a lot of places if we want to revert :(